dodo.c

```diff
@@ -743,8 +769,40 @@ int eval_byte(struct Program *p, struct Instruction *cur){
 }
 
 int eval_line(struct Program *p, struct Instruction *cur){
-    puts("eval_line unimplemented");
-    return 1; /* FIXME unimplemented */
+    char buffer[1024];
+    int observed = 0;
+    int i = 0;
+    size_t nread = 0;
+
+    /* first things first; seek to start of file */
+    if( fseek(p->file, 0, SEEK_SET) ){
+        puts("eval_line: fseek failed");
+        return 1;
+    }
+    p->offset = 0;
+
+    /* nothing more to be done if line 1 was requested */
+    if( cur->argument.num == 1 ){
+        return 0;
+    }
+
+    while( (nread = fread(buffer, 1, sizeof(buffer), p->file)) ){
+        for( i = 0; i < nread; i++ ){
```

Comment:
*very* minor style point, we should prefer `++i` when it doesn't matter (only using `i++` in the special case where we want the old value before the incr.)